test/lib/jdk/test/whitebox/WhiteBox.java

```diff
@@ -556,6 +558,31 @@ public void clearInlineCaches(boolean preserve_static_stubs) {
   // Force Full GC
   public native void fullGC();
 
+  private static Method wfrp = null;
+  /* Wait for reference processing, via Reference.waitForReferenceProcessing().
+   * Callers of this method will need the
+   * @modules java.base/java.lang.ref:open
+   * jtreg tag.
+   *
+   * This method should usually be called after a call to WhiteBox.fullGC().
+   */
+  public static void waitForReferenceProcessing() {
```

Comment:
Can the code duplication be avoided?

Reply:
That would be nice.
A primary goal here is to not disrupt existing tests that use `WhiteBox` or `ForceGC`, (e.g. by necessitating additional tags, `@run` arguments, etc).
I don't think there's a way for `WhiteBox.java` and `ForceGC.java` to both refer to code at some shared location without tests having to add a new tag - `@build`, I think.
However I will look into it a bit further.

Reply:
(No code duplication if just adding to WhiteBox.)